crates/version/build.rs

```diff
@@ -1,36 +1,56 @@
 use std::process::Command;
 
+use eyre::{eyre, Result as EyreResult};
+use rustc_version::version;
+
 fn main() {
-    let git_describe = Command::new("git")
-        .args([
+    if let Err(err) = try_main() {
+        eprintln!("build.rs error: {err}");
+        std::process::exit(1);
+    }
+}
+
+fn try_main() -> EyreResult<()> {
+    let git_describe = run_command(
+        "git",
+        &[
             "describe",
             "--always",
             "--dirty=-modified",
             "--tags",
             "--match",
             "[0-9]*",
-        ])
-        .output()
-        .ok()
-        .and_then(|o| String::from_utf8(o.stdout).ok())
-        .unwrap_or_else(|| "unknown".to_string());
+        ],
+    )?;
 
-    let git_commit = Command::new("git")
-        .args(["rev-parse", "--short", "HEAD"])
-        .output()
-        .ok()
-        .and_then(|o| String::from_utf8(o.stdout).ok())
-        .unwrap_or_else(|| "unknown".to_string());
+    let git_commit = run_command("git", &["rev-parse", "--short", "HEAD"])?;
+
+    let rustc_version = version()
```

Comment:
version is ambiguous, this whole crate is for versions

```suggestion
    let rustc_version = rustc_version::version()
```